pandapipes/component_models/water_tower_component.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2020 by Fraunhofer Institute for Energy Economics
+# and Energy System Technology (IEE), Kassel. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+import numpy as np
+from numpy import dtype
+
+from pandapipes.component_models.abstract_models import NodeElementComponent
+from pandapipes.constants import GRAVITATION_CONSTANT, P_CONVERSION
+from pandapipes.idx_branch import FROM_NODE, TO_NODE, LOAD_VEC_NODES
+from pandapipes.idx_node import PINIT, LOAD, NODE_TYPE, P, EXT_GRID_OCCURENCE
+from pandapipes.internals_toolbox import _sum_by_group
+from pandapipes.pipeflow_setup import get_lookup
+
+try:
+    import pplog as logging
+except ImportError:
+    import logging
+
+logger = logging.getLogger(__name__)
+
+
+class WaterTower(NodeElementComponent):
+    """
+    The water tower is similiar to an external grid. Based on the water column height the relative pressure
+    is determined. This pressure is fixed, i.e. the capacity of the water tower is infinitem and its level is constant.
+    """
+
+    @classmethod
+    def table_name(cls):
+        return "water_tower"
+
+    @classmethod
+    def sign(cls):
+        return 1.
+
+    @classmethod
+    def create_pit_node_entries(cls, net, node_pit, node_name):
+        """
+        Function which creates pit node entries.
+
+        :param net: The pandapipes network
+        :type net: pandapipesNet
+        :param node_pit:
+        :type node_pit:
+        :return: No Output.
+        """
+        water_towers = net[cls.table_name()]
+        density = net.fluid.get_density(water_towers.t_k.values)
+        junction_idx_lookups = get_lookup(net, "node", "index")[node_name]
+        junction = cls.get_connected_junction(net)
+        press = density * water_towers.height_m.values * \
+                GRAVITATION_CONSTANT / P_CONVERSION * water_towers.in_service.values
+        juncts_p, press_sum, number = _sum_by_group(junction.values, press,
+                                                    np.ones_like(press, dtype=np.int32))
```

Comment:
I am not sure whether this is physically possible. The [effect of communicating vessels](https://en.wikipedia.org/wiki/Communicating_vessels) (dt: ["kommunizierende Röhren"](https://de.wikipedia.org/wiki/Kommunizierende_R%C3%B6hren)) is not considered, is it?
I'd suggest limiting the number of water towers to 1 per net or raise a warning when towers with different water height are connected.